### DISCUSSION TITLE
Steps for baking FEL into BL

### DISCUSSION THREAD

**Discussion by mirobartovic** [R3S3t9999/DoVi_Scripts#14] [General]
Hi, I have a few DV movies that use FEL to expand the brightness. If I understand the issue correctly, I can’t do the simple Mode 7 conversion where the FEL of Profile 7 is discarded and BL+RPU is saved as Profile 8.1. I should bake in the FEL using DoviBaker instead and reencode the new BL. Unfortunately in the Mode M menu there’s no explicit “Bake FEL into BL” option. Should I use HDR to HDR encode and then generate the RPU again using Mode H HDR10 to DV conversion? Or am I overcomplicating things and the HDR10 to DV conversion uses also existing FEL metadata to generate the new RPU so it is enough to just do that?

**Comment by R3S3t9999** [comment]
Mode 7 is just for screenshots and in order to bake FEL into a new BL, it needs re-encoding with workflow 8-6-1.  
Everything is automated (encoding, rpu, injection ) but the script requires an Nvidia GPU.

**Comment by pickle1234d** [comment]
On this topic is there anyway we can use the GPU to encode or substitute X265 with NVenc when using the 8-6-1 workflow? I have quite a big library and doing it on the CPU at this scale would likely take many months!

Thanks

**Comment by R3S3t9999** [comment]
Yes, I guess I could add NVenc workflows. I dont know when I will check this out though.

**Comment by pickle1234d** [comment]
That would be amazing if you could! Thanks for all your work on this stuff!

**Comment by R3S3t9999** [comment]
@pickle1234d

Please try [this version](https://drive.google.com/file/d/128gq8aDUTKA_aT7SQsM9dkjA1EP1sosR/view?usp=drive_link) (it was fast and easy to implement). By default, it will encode with x265.exe so you have to change line 153 to set HEVC=NVENC
You must also add the [NVENC folder](https://mega.nz/file/BOVGHIha#hKXrdRXE86VVSPtvk1j0FzfHwMe6mLBXJQPGGK3MWVg) and the encode settings files [SDR](https://mega.nz/file/pLkQgSaD#InStT2LXf2rAxUszjpJtLzlzpDbkdz6w_S4oJSxRE0c) and [HDR](https://mega.nz/file/kGlRWKjK#XXpu1DTs_eR8Rt7NDIcZVbRT_H03mbwmR54wdt4p5JU)  in the script tools folder.

The default encode settings are VBR 2-pass full, preset 7 at 50 000kbps. Those settings produces quality close to default x265.exe CRF-15 slow.
https://slow.pics/c/uWYbNVdC

**Reply by pickle1234d** [reply]
Just testing this out now and the render time has dropped significantly so thank you! will have a play with bitrates and see what works best for my setup in time.  

Thank you so much for implementing ill be donating shortly =]

**Comment by Gag21** [comment]
Hi, I am getting this error when using NVENC even with latest versions

```
bitstream: Setting masterdisplay/contentlight not supported in H.264/AVC encoding.
failed to initialize file reader(s).
Failed to open output file: "E:\Movies\DoVi_Scripts\00294_encoded.hevc"
```

x265 is working but at 1fps it is unusable for me. 
I have a RTX3070 with latest drivers.

**Comment by R3S3t9999** [comment]
I dont know, it's working on my end.  I also have a rtx3070.
Can you set the script to ''echo on'' (first line with notepad) and post the full log please?

https://justpaste.it/a5z76

**Comment by Gag21** [comment]
https://jpst.it/3HFYP

**Reply by R3S3t9999** [reply]
try the [latest version](https://github.com/R3S3t9999/DoVi_Scripts/releases)

**Reply by Gag21** [reply]
yup working now thanks you are quick.

**Reply by Gag21** [reply]
getting 10fps now and my poor cpu still pegged at 100% do you think cbr will reduce cpu load? also what bitrates to use to get close to 100 Mb/s 4k bd source quality.

**Reply by R3S3t9999** [reply]
I dont know, i dont use NVenc so you'll have to experiment.
